test_util/testutil.h

```diff
@@ -429,6 +429,32 @@ class SleepingBackgroundTask {
       bg_cv_.Wait();
     }
   }
+  // Return true if after a short wait, the task ends up with sleeping
+  bool ShortWaitForSleeping() {
+    MutexLock l(&mutex_);
+    // 1000us * 10000 = 10 sec max
+    for (int i = 0; i < 10000; i++) {
+      if (sleeping_) {
+        return true;
+      }
+      const uint64_t kWaitDurationMicros = 1000;
+      bg_cv_.TimedWait(Env::Default()->NowMicros() + kWaitDurationMicros);
```

Comment:
Is there any particular reason for TimedWait instead of Wait?

Reply:
The reason is that, in case there is a bug and the condition never satisfied, I don't want the test fails as timing out, but prints out which condition isn't met.

Reply:
SGTM. Failing assertion early is better than timeout.